internal/client/policy/policy_resource.go

```diff
@@ -0,0 +1,263 @@
+
+	ManageV1alpha1OrganizationPolicyResourceServiceCreate(request *policyorganizationmodel.VmwareTanzuManageV1alpha1OrganizationPolicyPolicyRequest) (*policyorganizationmodel.VmwareTanzuManageV1alpha1OrganizationPolicyPolicyResponse, error)
+
+	ManageV1alpha1OrganizationPolicyResourceServiceDelete(fn *policyorganizationmodel.VmwareTanzuManageV1alpha1OrganizationPolicyFullName) error
+
+	ManageV1alpha1OrganizationPolicyResourceServiceGet(fn *policyorganizationmodel.VmwareTanzuManageV1alpha1OrganizationPolicyFullName) (*policyorganizationmodel.VmwareTanzuManageV1alpha1OrganizationPolicyGetPolicyResponse, error)
+
+	ManageV1alpha1OrganizationPolicyResourceServiceUpdate(request *policyorganizationmodel.VmwareTanzuManageV1alpha1OrganizationPolicyPolicyRequest) (*policyorganizationmodel.VmwareTanzuManageV1alpha1OrganizationPolicyPolicyResponse, error)
+
+	ManageV1alpha1WorkspacePolicyResourceServiceCreate(request *policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyPolicyRequest) (*policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyPolicyResponse, error)
+
+	ManageV1alpha1WorkspacePolicyResourceServiceDelete(fn *policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyFullName) error
+
+	ManageV1alpha1WorkspacePolicyResourceServiceGet(fn *policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyFullName) (*policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyGetPolicyResponse, error)
+
+	ManageV1alpha1WorkspacePolicyResourceServiceUpdate(request *policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyPolicyRequest) (*policyworkspacemodel.VmwareTanzuManageV1alpha1WorkspacePolicyPolicyResponse, error)
+}
+
+/*
+ManageV1alpha1ClusterPolicyResourceServiceCreate creates a policy.
+*/
+func (p *Client) ManageV1alpha1ClusterPolicyResourceServiceCreate(request *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyRequest) (*policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyResponse, error) {
+	requestURL := fmt.Sprintf("%s/%s/%s", "v1alpha1/clusters", request.Policy.FullName.ClusterName, "policies")
+	policyClusterResponse := &policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyPolicyResponse{}
+	err := p.Create(requestURL, request, policyClusterResponse)
+
+	return policyClusterResponse, err
+}
+
+/*
+ManageV1alpha1ClusterPolicyResourceServiceDelete deletes a policy.
+*/
+func (p *Client) ManageV1alpha1ClusterPolicyResourceServiceDelete(fn *policyclustermodel.VmwareTanzuManageV1alpha1ClusterPolicyFullName) error {
+	queryParams := url.Values{}
+
+	if fn.ManagementClusterName != "" {
+		queryParams["fullName.managementClusterName"] = []string{fn.ManagementClusterName}
+	}
+
+	if fn.ProvisionerName != "" {
+		queryParams["fullName.provisionerName"] = []string{fn.ProvisionerName}
+	}
+
+	requestURL := fmt.Sprintf("%s/%s/%s/%s?%s", "v1alpha1/clusters", fn.ClusterName, "policies", fn.Name, queryParams.Encode())
```

Comment:
org ID also goes as a query parameter and the policy name as path variable

Reply:
This is as per the generated client code. Did not notice any org id in that. As for the policy name, it is already in the path variable.

Reply:
I think it is not from the generated client. As per the API's we do need the org ID. It is not being set as per the changes here

Reply:
From the generated code:

PathPattern:        "/v1alpha1/clusters/{fullName.clusterName}/policies/{fullName.name}",
		
The org id is optional like in cluster resource service API and is substituted from the auth token.

/* ManageV1alpha1ClusterPolicyResourceServiceDeleteParams contains all the parameters to send to the API endpoint
   for the manage v1alpha1 cluster policy resource service delete operation.

   Typically these are written to a http.Request.
*/
type ManageV1alpha1ClusterPolicyResourceServiceDeleteParams struct {

	/* FullNameClusterName.

	   Name of the cluster.
	*/
	FullNameClusterName string

	/* FullNameManagementClusterName.

	   Name of the management cluster.
	*/
	FullNameManagementClusterName *string

	/* FullNameName.

	   Name of the policy.
	*/
	FullNameName string

	/* FullNameOrgID.

	   ID of Organization.
	*/
	FullNameOrgID *string

	/* FullNameProvisionerName.

	   Name of the cluster provisioner.
	*/
	FullNameProvisionerName *string

	timeout    time.Duration
	Context    context.Context
	HTTPClient *http.Client
}